Quick_Deploy/HuggingFaceTransformers/README.md

```diff
@@ -0,0 +1,263 @@
+- [adept/persimmon-8b-base](https://huggingface.co/adept/persimmon-8b-base)
+
+These models were selected because of their popularity and consistent response quality.
+However, this tutorial is also generalizable for any transformer model provided
+sufficient infrastructure.
+
+*NOTE*: The tutorial is intended to be a reference example only. It may not be tuned for
+optimal performance.
+
+## Step 1: Create a Model Repository
+
+The first step is to create a model repository containing the models we want the Triton
+Inference Server to load and use for inference processing. To accomplish this, create a
+directory called `model_repository` and copy the `falcon7b` model folder into it:
+
+```
+mkdir -p model_repository
+cp -r falcon7b/ model_repository/
+```
+
+The `falcon7b/` folder we copied is organized in the way Triton expects and contains
+two important files needed to serve models in Triton:
+- **config.pbtxt** - Outlines the backend to use, model input/output details, and custom
+parameters to use for execution. More information on the full range of model configuration
+properties Triton supports can be found [here](https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/user_guide/model_configuration.html).
+- **model.py** - Implements how Triton should handle the model during the initialization,
+execution, and finalization stages. More information regarding python backend usage
+can be found [here](https://github.com/triton-inference-server/python_backend#usage).
+
+
+## Step 2: Build a Triton Container Image
+
+The second step is to create an image that includes all the dependencies necessary
+to deploy hugging face transformer models on the Triton Inference Server. This can be done
+by building an image from the provided Dockerfile:
+
+```
+docker build -t triton_transformer_server .
+```
+
+## Step 3: Launch the Triton Inference Server
+
+Once the ```triton_transformer_server``` image is created, you can launch the Triton Inference
+Server in a container with the following command:
+
+```bash
+docker run --gpus all -it --rm --net=host --shm-size=1G --ulimit memlock=-1 --ulimit stack=67108864 -v ${PWD}/model_repository:/opt/tritonserver/model_repository triton_transformer_server tritonserver --model-repository=model_repository
+```
+
+The server has launched successfully when you see the following outputs in your console:
+
+```
+I0922 23:28:40.351809 1 grpc_server.cc:2451] Started GRPCInferenceService at 0.0.0.0:8001
+I0922 23:28:40.352017 1 http_server.cc:3558] Started HTTPService at 0.0.0.0:8000
+I0922 23:28:40.395611 1 http_server.cc:187] Started Metrics Service at 0.0.0.0:8002
+```
+
+## Step 4: Query the Server
+
+Now we can query the server using curl, specifying the server address and input details:
+
+```json
+curl -X POST localhost:8000/v2/models/falcon7b/infer -d '{"inputs": [{"name":"text_input","datatype":"BYTES","shape":[1],"data":["I am going"]}]}'
```

Comment:
We can also curl with the `/generate` endpoint now. Feel free to add in a separate PR if you want though.

Reply:
ref: https://github.com/triton-inference-server/server/blob/main/docs/protocol/extension_generate.md

Reply:
I'll create a minor ticket for this and wait for the 23.10 container to be released.